[ADD] accounting: l10n_br avalara implementation

samueljlieber opened the pull request:
This PR adds the [Brazilian localization Avalara user implementation documentation](https://docs.google.com/document/d/1Un4XhU6PQIetf2Q5et3SekQepQlZMU5rxeliTswoGwc/edit).

Task: [3459876](https://www.odoo.com/web#id=3459876&cids=3&menu_id=4720&action=333&active_id=3835&model=project.task&view_type=form)

robodoo commented:
[Pull request status dashboard](https://mergebot.odoo.com/odoo/documentation/pull/5199).

vbe-odoo commented on content/applications/finance/fiscal_localizations/brazil/api-call-triggers.png:
I agree, here is a text version of the digram:

### **API Calls Triggered by Actions**
This guide provides an overview of the actions that trigger API calls for tax computation. Please note that each API call incurs a cost.

**API Calls Triggered in Sale Order / Quotation**
The following actions performed on a Sale Order or Quotation will trigger the API call for tax computation:
* Manual Trigger: Users can manually initiate the tax computation by clicking on "Compute Taxes Using Avatax".
* Quotation Confirmation: The API call is automatically triggered when a Quotation is confirmed and transformed into a Sale Order.
* Preview: Clicking on the "Preview" button will trigger the API call.
* Emailing Quotation / Sale Order: The API call is triggered when a Quotation or Sale Order is sent to a customer via email.
* Online Quotation Access: When a customer accesses the Quotation online (via the Portal View), the API call is triggered.

**API Calls Triggered in Invoice**
* The following actions performed on an Invoice will trigger the API call for tax computation:
* Manual Trigger: Users can manually initiate the tax computation by clicking on "Compute Taxes Using Avatax".
* Preview: Clicking on the "Preview" button will trigger the API call.
* Online Invoice Access: When a customer accesses the Invoice online (via the Portal View), the API call is triggered.

Please remember that each API call has an associated cost. Be mindful of the actions that trigger these calls to manage costs effectively.

vbe-odoo commented on content/applications/finance/fiscal_localizations/brazil/api-call-triggers-glossary.png:
@samueljlieber and we can remove this image, as with the text version this one won't be needed.
Thanks!

samueljlieber commented:
Implemented @vbe-odoo's [API Trigger Action content](https://github.com/odoo/documentation/pull/5199#discussion_r1273905317) in https://github.com/odoo/documentation/commit/280562c86fe5e78778f3b9f52dbfa6f84e4ba6e6.

vbe-odoo commented on content/applications/finance/fiscal_localizations/brazil.rst:
```diff
@@ -0,0 +1,371 @@
+       chart of accounts, taxes, and include required fields to properly configure the contact.
+   * - :guilabel:`Brazil - Accounting Reports`
+     - `l10n_br_reports`
+     - Includes all the technical and functional requirements to generate and validate
+       :doc:`electronic documents
+       </applications/finance/accounting/customer_invoices/electronic_invoicing>`, based on the
+       technical documentation published by the SRI. The authorized documents are: invoices, credit
+       notes, debit notes, withholdings and purchase liquidations.
```
```suggestion
       </applications/finance/accounting/customer_invoices/electronic_invoicing>`, adds a simple Tax Report that helps check the tax amount per tax group in a given period of time. Also adds the P&L and BS adapted for the Brazilian market.
```

Can you help with this correction @samueljlieber as I made a mistake and kept with the wrong info here.
Thanks!

vbe-odoo replied:
```suggestion
     - Adds a simple Tax Report that helps check the tax amount per tax group in a given period of time. Also adds the P&L and BS adapted for the Brazilian market.
```

Can you help with this correction @samueljlieber as I made a mistake and kept with the wrong info here.
Thanks!

vbe-odoo commented on content/applications/finance/fiscal_localizations/brazil.rst:
```diff
@@ -0,0 +1,371 @@
+Computation` to :guilabel:`Percentage of Price Tax Included`, and on the :guilabel:`Advanced
+Options` tab check the :guilabel:`Included in Price` option.
+
+.. image:: brazil/tax-configuration.png
```
@samueljlieber also this image is wrong, I retook the screenshot.
Here is the one to be used here: https://drive.google.com/file/d/1w0JnmUjeY_iycauVkPyYUZ90bKKpehbK/view?usp=sharing

Sorry for that.

vbe-odoo commented on content/applications/finance/fiscal_localizations/brazil.rst:
```diff
@@ -0,0 +1,371 @@
+may differ, and **taxes attached to services will not be computed by Avatax**, only good taxes are
+computed.
+
+When adding a tax used for a service that is included in the final price - when the tax is not added
+or subtracted on top of the product price - then you need to configure the :guilabel:`Tax
+Computation` to :guilabel:`Percentage of Price Tax Included`, and on the :guilabel:`Advanced
+Options` tab check the :guilabel:`Included in Price` option.
```
```suggestion
When adding a tax used for a service that is included in the final price - when the tax is not added
or subtracted on top of the product price - then you need to configure the :guilabel:`Tax
Computation` to :guilabel:`Percentage of Price Tax Included`, and on the :guilabel:`Advanced
Options` tab check the :guilabel:`Included in Price` option.

For more information on how to configure taxes to better fit your needs, please go to the [taxes functional documentation](https://www.odoo.com/documentation/16.0/applications/finance/accounting/taxes.html#taxes)
```

vbe-odoo replied:
@samueljlieber please help me add this also here.
Thanks!

vbe-odoo reviewed (commented):
@samueljlieber I left some changes.
Besides those changes, the rest looks good to me 🥳 
Thanks!

vbe-odoo commented on content/applications/finance/fiscal_localizations/brazil.rst:
```diff
@@ -0,0 +1,371 @@
+  :alt: Company fiscal configuration.
+
+Finally, upload a company logo and save the contact.
+
```
@samueljlieber can we also add an additional Note here below saying:
If you are a Simplified Regime you need to configure the ICMS rate in :menuselection:`Accounting
--> Configuration --> Settings --> Taxes --> Avatax Brazil` 

Thanks!

samueljlieber commented:
Thank you @vbe-odoo for the review! I implemented your changes in https://github.com/odoo/documentation/commit/45d7630d5e0b0f0fe7d3f62ba6d3ac0569533922. I believe this PR should be all set for https: @odoo/localizations-doc-review 🙂

dade-odoo commented:
Hi @vbe-odoo and @samueljlieber, thanks for adding to the growing list of fiscal localization documentation! I'm sure this one will be used a lot, especially with the new office :)

Instead of leaving lots of comments, it was easier for me to create a separate commit. Of course, then I managed to accidentally create two commits, and when I tried to squash them together I ended up squashing mine with Sam's. Oops. Let me know if you have any questions! Please review and let me know if you disagree with any of my changes.

In general, I made changes to:

- reference other documents or sections of this document where appropriate
- improve formatting
- shorten the doc by consolidating repetitive sections and removing unnecessary screenshots

samueljlieber commented:
Hi @dade-odoo, thanks for the improvements! These changes look good to me, @vbe-odoo please let us know if you have any changes or comments 🙂 Thanks all!

vbe-odoo commented:
Hi @dade-odoo thanks for the modifications.
I've taken a look and it looks good to me!
Also, thanks @samueljlieber for taking a look at it!
🚀

xpl-odoo commented on content/applications/finance/fiscal_localizations/brazil.rst:
```diff
@@ -0,0 +1,327 @@
+   :alt: Avatax account configuration.
+
+.. note::
+   You can transfer API credentials. Use this only when you already created an account in another
+   Odoo instance and wish to reuse it.
```
```suggestion
   You can transfer API credentials. Use this only when you have already created an account in another
   Odoo instance and wish to reuse it.
```

xpl-odoo commented on content/applications/finance/fiscal_localizations/brazil.rst:
```diff
@@ -0,0 +1,327 @@
+~~~~~~~~~~~~~~~~~
+
+The :doc:`chart of accounts </applications/finance/accounting/get_started/chart_of_accounts>` is
+installed by default as part of the set of data included in the localization module. The accounts
```
```suggestion
installed by default as part of the data set included in the localization module. The accounts
```

xpl-odoo commented on content/applications/finance/fiscal_localizations/brazil.rst:
```diff
@@ -0,0 +1,327 @@
+Taxes are automatically created when installing the Brazilian localization. Taxes are already
+configured, and some of them are used by Avalara when computing taxes on the sales order or invoice.
+
+Taxes can be edited or more taxes can be added. For example, some taxes used for services need to be
```
```suggestion
Taxes can be edited, or more taxes can be added. For example, some taxes used for services need to be
```

xpl-odoo commented on content/applications/finance/fiscal_localizations/brazil.rst:
```diff
@@ -0,0 +1,327 @@
+:guilabel:`Percentage of Price Tax Included`, and, on the :guilabel:`Advanced Options` tab, check
+the :guilabel:`Included in Price` option.
+
+For more information on how to configure taxes to better fit your needs, please go to the
```
```suggestion
For more information on configuring taxes to fit your needs better, please go to the
```

xpl-odoo commented on content/applications/finance/fiscal_localizations/brazil.rst:
```diff
@@ -0,0 +1,327 @@
+To use the AvaTax integration on sale orders and invoices, first specify the following information
+on the product:
+
+- :guilabel:`CEST Code`
+- :guilabel:`Mercosul NCM Code`
+- :guilabel:`Source of Origin`
+- :guilabel:`SPED Fiscal Product Type`
+- :guilabel:`Purpose of Use`
```
More info on filling out these fields could be useful

xpl-odoo commented on content/applications/finance/fiscal_localizations/brazil.rst:
```diff
@@ -0,0 +1,327 @@
+
+.. note::
+   Odoo automatically creates three products to be used for transportation costs associated with
+   sales. These are named `Freight`, `Insurance`, and `Other Costs`, they are already configured, if
+   more need to be created, duplicate and use the same configuration (configuration needed:
```
```suggestion
   sales. These are named `Freight`, `Insurance`, and `Other Costs`. They are already configured, and if
   more need to be created, duplicate and use the same configuration (configuration needed:
```

xpl-odoo commented on content/applications/finance/fiscal_localizations/brazil.rst:
```diff
@@ -0,0 +1,327 @@
+#. General information about the contact:
+
+   - Select the :guilabel:`Company` option for a contact with a tax ID (CNPJ), or check
+     :guilabel:`Individual` for a contact with a CPF
```
```suggestion
     :guilabel:`Individual` for a contact with a CPF.
```

xpl-odoo commented on content/applications/finance/fiscal_localizations/brazil.rst:
```diff
@@ -0,0 +1,327 @@
+   - Select the :guilabel:`Company` option for a contact with a tax ID (CNPJ), or check
+     :guilabel:`Individual` for a contact with a CPF
+   - :guilabel:`Name`
+   - :guilabel:`Address`: :guilabel:`Zip Code` is a required field to properly compute taxes.
```
```suggestion
   - :guilabel:`Address`: :guilabel:`Zip Code` is a required field to compute taxes properly.
```

xpl-odoo commented on content/applications/finance/fiscal_localizations/brazil.rst:
```diff
@@ -0,0 +1,327 @@
+     :guilabel:`Individual` for a contact with a CPF
+   - :guilabel:`Name`
+   - :guilabel:`Address`: :guilabel:`Zip Code` is a required field to properly compute taxes.
+   - :guilabel:`Tax ID` or :guilabel:`CPF`: depending on the case complete with the number
```
unclear

vbe-odoo replied:
```suggestion
   - :guilabel:`Tax ID` or :guilabel:`CPF`: Use CPF for individuals and Tax ID (CNPJ) for companies
```
Also, in BR users know what these abbreviations refer to, but true that this is unclear.
@samueljlieber can you help me with this change?
Thanks!

xpl-odoo commented on content/applications/finance/fiscal_localizations/brazil.rst:
```diff
@@ -0,0 +1,327 @@
+
+#. Fiscal information about the contact under the :guilabel:`Sales & Purchase` tab:
+
+   - :guilabel:`Fiscal Position`: add the AvaTax fiscal position to automatically compute taxes on
+     sale orders and invoices
```
```suggestion
   - :guilabel:`Fiscal Position`: add the AvaTax fiscal position to compute taxes on
     sale orders and invoices automatically.
```

xpl-odoo commented on content/applications/finance/fiscal_localizations/brazil.rst:
```diff
@@ -0,0 +1,327 @@
+     sale orders and invoices
+   - :guilabel:`Tax Regime`: federal tax regime
+   - :guilabel:`ICMS Taxpayer Type`: taxpayer type determines if the contact is within the ICMS
+     regime, if it is exempt, or if it is a non-taxpayer
```
```suggestion
     regime, if it is exempt, or if it is a non-taxpayer.
```

xpl-odoo commented on content/applications/finance/fiscal_localizations/brazil.rst:
```diff
@@ -0,0 +1,327 @@
+~~~~~~~~~~~~~~~~
+
+To compute taxes on sale orders and invoices, it is necessary to have a :guilabel:`Fiscal Position`
+with the :guilabel:`Detect Automatically` and the :guilabel:`Use AvaTax API` selected.
```
```suggestion
with the :guilabel:`Detect Automatically` and the :guilabel:`Use AvaTax API` options enabled.
```

xpl-odoo commented on content/applications/finance/fiscal_localizations/brazil.rst:
```diff
@@ -0,0 +1,327 @@
+Tax calculations on quotation / sales orders
+--------------------------------------------
+
+Trigger an API call to calculate taxes on a quotation or sales order automatically with AvaTax any
```
```suggestion
Trigger an API call to calculate taxes on a quotation or sales order automatically with AvaTax in any
```

xpl-odoo commented on content/applications/finance/fiscal_localizations/brazil.rst:
```diff
@@ -0,0 +1,327 @@
+
+.. note::
+   The :guilabel:`Fiscal Position` must be set to `Automatic Tax Mapping (Avalara Brazil)` for any
+   of these actions to automatically compute taxes.
```
```suggestion
   of these actions to compute taxes automatically.
```

xpl-odoo reviewed (changes requested):
Nice job! Most of my suggestions concern small grammar issues.
Note for screenshots: generally, we don't need the full-width form when we only need to highlight a part of it. For example, this is sufficient:
![image](https://github.com/odoo/documentation/assets/70884563/684562ef-22a9-4d96-8218-be6505c898b6)

samueljlieber commented:
Thank you for the thorough review @xpl-odoo! I implemented all of the currently resolved suggestions in https://github.com/odoo/documentation/commit/30dddd337d22ab61163df0bd0c6cbf8a5b845abd and @vbe-odoo I'll need help with the remaining 🙏:

- [Maybe clarify what these fields are](https://github.com/odoo/documentation/pull/5199#discussion_r1285603528)
- [Maybe clarify what these fields are ](https://github.com/odoo/documentation/pull/5199#discussion_r1285606710)
- [under the simplified regime, no?](https://github.com/odoo/documentation/pull/5199#discussion_r1285607942)
- [Why not include the (Avalara portal) instructions found in the link in the documentation?](https://github.com/odoo/documentation/pull/5199#discussion_r1285755018)
- [More info on filling out these fields could be useful](https://github.com/odoo/documentation/pull/5199#discussion_r1285775245)
- [Tax ID or CPF unclear](https://github.com/odoo/documentation/pull/5199#discussion_r1285780302)

Thank you!

vbe-odoo commented on content/applications/finance/fiscal_localizations/brazil.rst:
```diff
@@ -0,0 +1,327 @@
+   - :guilabel:`Name`
+   - :guilabel:`Address` (add :guilabel:`City`, :guilabel:`State`, :guilabel:`Zip Code`,
+     :guilabel:`Country`)
+   - Tax ID (:guilabel:`CNPJ`)
+   - :guilabel:`IE`
+   - :guilabel:`IM`
+   - :guilabel:`SUFRAMA code` (if applicable)
```
They are common knowledge in BR.
But we can expand the abbreviations:
1. IE (State Registration)
2. IM (Municipal Registration)
3. SUFRAMA (Superintendence of the Manaus Free Trade Zone)

vbe-odoo commented on content/applications/finance/fiscal_localizations/brazil.rst:
```diff
@@ -0,0 +1,315 @@
+   - :guilabel:`Address` (add :guilabel:`City`, :guilabel:`State`, :guilabel:`Zip Code`,
+     :guilabel:`Country`)
+   - Tax ID (:guilabel:`CNPJ`)
+   - :guilabel:`IE`
+   - :guilabel:`IM`
+   - :guilabel:`SUFRAMA code` (if applicable)
```
```suggestion
   - :guilabel:`IE` (State Registration)
   - :guilabel:`IM` (Municipal Registration)
   - :guilabel:`SUFRAMA code` (Superintendence of the Manaus Free Trade Zone - if applicable)
```

@samueljlieber maybe something like this. Please feel free to improve it if needed

vbe-odoo commented on content/applications/finance/fiscal_localizations/brazil.rst:
```diff
@@ -0,0 +1,327 @@
+#. Configure the :guilabel:`Fiscal Information` within the :guilabel:`Sales and Purchase` tab:
+
+   - Add the :guilabel:`Fiscal Position` for :ref:`Avatax Brazil <brazil/fiscal-positions>`.
+   - :guilabel:`Tax Regime`
+   - :guilabel:`ICMS Taxpayer Type`
+   - :guilabel:`Main Activity Sector`
```
```suggestion
   - :guilabel:`Tax Regime` (Federal Tax Regime)
   - :guilabel:`ICMS Taxpayer Type` (indicates ICMS regime, Exempt status, or Non-Taxpayer.)
   - :guilabel:`Main Activity Sector`
```

@samueljlieber here are some suggestions, please feel free to improve it if needed.
For the last one "Main Activity Sector" the label is already the description of what it is, so no further clarifications are needed there)

vbe-odoo commented on content/applications/finance/fiscal_localizations/brazil.rst:
```diff
@@ -0,0 +1,327 @@
+#. Finally, upload a company logo and save the contact.
+
+.. note::
+   If you are a simplified regime you need to configure the ICMS rate in
```
Yes. agree with this change.
Let's use "_under_" instead of "_in_".
@samueljlieber can you apply this change?

vbe-odoo commented:
> Thank you for the thorough review @xpl-odoo! I implemented all of the currently resolved suggestions in [30dddd3](https://github.com/odoo/documentation/commit/30dddd337d22ab61163df0bd0c6cbf8a5b845abd) and @vbe-odoo I'll need help with the remaining 🙏:
> 
> * [Maybe clarify what these fields are](https://github.com/odoo/documentation/pull/5199#discussion_r1285603528)
> * [Maybe clarify what these fields are ](https://github.com/odoo/documentation/pull/5199#discussion_r1285606710)
> * [under the simplified regime, no?](https://github.com/odoo/documentation/pull/5199#discussion_r1285607942)
> * [Why not include the (Avalara portal) instructions found in the link in the documentation?](https://github.com/odoo/documentation/pull/5199#discussion_r1285755018)
> * [More info on filling out these fields could be useful](https://github.com/odoo/documentation/pull/5199#discussion_r1285775245)
> * [Tax ID or CPF unclear](https://github.com/odoo/documentation/pull/5199#discussion_r1285780302)
> 
> Thank you!

Hey @samueljlieber thanks for addressing the rest of the comments!
I've added my comments and suggestions to each of the comments you've listed!

Please help me add the suggestions - or improve them - and let me know if there is something else needed.
Thanks!

samueljlieber commented:
Thank you @vbe-odoo! I implemented your improvements in https://github.com/odoo/documentation/commit/deedd5f094d9461092f9ed0727dfb0548e5cc741. @xpl-odoo we have made the changes you suggested and I corrected a couple early line breaks. All set for a second look 🙏 thank you!

xpl-odoo commented on content/applications/finance/fiscal_localizations/brazil.rst:
```diff
@@ -0,0 +1,327 @@
+   <https://drive.google.com/file/d/1jPtKzilfLrfdRXyPGacQNr6i2zvQvoOd/view?usp=sharing>`_. You can
+   start using AvaTax in Odoo without creating a password and accessing the Avalara Portal; for
+   Odoo, the only requirement to start using the Avalara Tax Computation Engine is to create an
+   account from the settings page.
```
It doesn't matter if something has to be configured outside of Odoo; we can - and sometimes should - still document processes outside of Odoo. Unless the external 'provider' documents it better (assuming here it would be better maintained) and/or it isn't necessary to configure the feature (i.e., extra info).
If the information makes too much noise, we don't need a link to an external document.

Here the information comes from us, meaning we could put it in the documentation instead. Furthermore, the information is relatively short at its core (the screenshots aren't needed):
> 1. Login on the [Avalara portal](https://portal.avalarabrasil.com.br/Login)
> 2. Click on “Meu primeiro acesso”
> 3. Add the email address you used in Odoo to create the Avalara/Avatax Account, and then click "Solicitar Senha"
> 4. You will receive an email with a Token and a link to create your password. Click on this link and copy-paste the token and allocate your desired password.

If you feel this information is superfluous, feel free not to include it, but I would avoid linking to an inconvenient and no so good-looking PDF.

samueljlieber commented:
Added the Avalara portal instructions directly into the documentation and removed the link to the PDF in https://github.com/odoo/documentation/commit/bddafb988b85165341a218c083ccae357668a2c6. Thank you everyone!

vbe-odoo commented:
> 

Thanks for the changes, LGTM!
Bests!

xpl-odoo commented:
@robodoo r+
